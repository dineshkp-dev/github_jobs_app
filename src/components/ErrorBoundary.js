import React, { Component } from 'react';
import '../styles/error.css';
import { HOME_PAGE_LINK } from '../common/constants';

export class ErrorBoundary extends Component {
  state = { error: false };

  static getDerivedStateFromError(error) {
    return { error };
  }

  render() {
    const errorMessage =
      this.state.error && this.state.error.message
        ? this.state.error.message
        : 'Oops! An Error has occurred.';
    return this.state.error ? (
      <div className='error-container'>
        <h1 className='error-title'>Error!</h1>
        <div className='error-msg'>
          <div className='error-text'>{errorMessage}</div>
          <div>
            Please click <a href={HOME_PAGE_LINK}>here</a> to go back to the
            home page
          </div>
        </div>
      </div>
    ) : (
      this.props.children
    );
  }
}

export default ErrorBoundary;
