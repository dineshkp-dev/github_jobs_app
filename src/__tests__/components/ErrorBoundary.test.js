import ErrorBoundary from '../../components/ErrorBoundary';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('ErrorBoundary snapshot test', () => {
  test('renders', () => {
    const { container } = render(
      <ErrorBoundary children={<div>dummy component</div>} />
    );
    expect(container).toMatchSnapshot();
  });
});
