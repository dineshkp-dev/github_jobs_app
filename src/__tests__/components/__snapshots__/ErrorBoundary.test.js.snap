// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary snapshot test renders 1`] = `
<div>
  <div>
    dummy component
  </div>
</div>
`;
